components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/internal/service/impl/UserAccountRecoveryManager.java

```diff
@@ -148,6 +160,57 @@ private void checkAccountLockedStatus(User user) throws IdentityRecoveryExceptio
         }
     }
 
+    /**
+     * Check whether the user account is eligible for account recovery.
+     *
+     * @param eventName                    Name of the event.
+     * @param user                         The user.
+     * @param recoveryScenario             Account recovery scenario.
+     * @param recoveryNotificationChannels Notification channel.
+     * @param metaProperties               Meta details.
+     * @throws IdentityRecoveryException If account doesn't satisfy the conditions to recover.
+     */
+    private void checkUserValidityForAccountRecovery(String eventName, User user, RecoveryScenarios recoveryScenario,
```

Comment:
We can move eventName inside this function. I see no need to pass it as an argument.

Reply:
fixed with 22b35544bed1f4fd15e8ff5f5e59a19c5e277b58